### ISSUE TITLE
Cannot list emails because of one email

### ISSUE THREAD

**Issue by chopin2712** [soywod/himalaya#401]
I just ran `himalaya list` and got the following output:

```
Error: cannot parse envelope

Caused by:
    cannot get sender's mailbox of message 754
```

so I couldn't list my inbox.

Once I removed that message, I got the same issue with message 751. Before removing them I exported the two messages in raw format, so I can still retrieve some data from those messages if that can help troubleshooting this.

**Comment by soywod** [comment]
Could you share (if it is not sensitive data) the shape of the sender's email address of those 2 emails? It looks like the left part cannot be parsed for some reason.

**Comment by chopin2712** [comment]
> Could you share (if it is not sensitive data) the shape of the sender's email address of those 2 emails? It looks like the left part cannot be parsed for some reason.

I am really sorry, I wasn't on GitHub for a while. It comes from the mailing list of tilde.team. It's part of a mailing list, so I don't know if that person will be OK with sending their email address in public. But if you have another mean of contact I guess I could give you more information.

All I can say here is that the url is hosted on `tilde.team` server and contains a letter then a few numbers.

**Comment by chopin2712** [comment]
OK just tired to list some things from the mail headers. Apparently, there is one address that makes the thing buggy. 

The only thing I noticed so far is the fact that it has a `Message-Id` while on any other mail I would find `Message-ID` also that `Message-Id` value had a value such as `something@grinningface` with no extension.

But all their mails are always buggy. Also I moved them into another mailbox and now when I try to read them it gives me this:

```
snowcode@snowcode:~$ himalaya -m School read 1
Error: cannot parse address "Address { name: None, adl: None, mailbox: None, host: None }"

Caused by:
    cannot get address mailbox
```
